spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/PublisherCallbackChannelImpl.java

```diff
@@ -658,7 +659,7 @@ private synchronized void generateNacksForPendingAcks(String cause) {
 				processAck(confirmEntry.getKey(), false, false, false);
 				iterator.remove();
```

Comment:
Not related to the PR, but this `remove()` is redundant: we `this.pendingConfirms.clear();` in the end of the method anyway.